sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -118,7 +118,7 @@ private[sql] object CatalogV2Implicits {
 
   implicit class MultipartIdentifierHelper(parts: Seq[String]) {
     if (parts.isEmpty) {
-      throw new AnalysisException("multi-part identifier cannot be empty.")
+      throw new AnalysisException("Namespaces in V1 catalog can have only a single name part.")
```

Comment:
to be more precise: `Namespaces in V1 catalog cannot be empty`?

Reply:
Actually, `parts` includes a table name. When we say that `parts` cannot be empty, we require at least a table name.

Probably, `Namespaces in V1 catalog can have only a single name part` could confuse users too.

We should say something like a table identifier must contain either a table name or database + a table name.

Specifically in the check, we should say **"Table identification must have at least a table name"**

Reply:
BTW, how does `SHOW TABLES IN $catalog` get related to table identifier?

Reply:
You mean `SHOW TABLES IN spark_catalog` since we are focusing on v1 only, right.

1. We cut off `spark_catalog` from namespaces in `SessionCatalogAndNamespace`, so, `ns == Seq.empty` here: https://github.com/apache/spark/blob/0617dfce7beb34662ab30a607721e9b46e65c21e/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L365
2. Then `ns.length != 1` is `true` and we try to raise the exception at https://github.com/apache/spark/blob/0617dfce7beb34662ab30a607721e9b46e65c21e/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L367
3. `ns.quoted` triggers implicit wrapping `Seq.empty` by `MultipartIdentifierHelper`, and fit hit the second check `if (parts.isEmpty)` at https://github.com/apache/spark/blob/156704ba0dfcae39a80b8f0ce778b73913db03b2/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala#L120-L122

Reply:
I think this is better solution: https://github.com/apache/spark/pull/30963